<launch>
  <arg name="camera" default="camera_flex_wide/left" />
  <arg name="show_images" default="false" />
  <arg name="models" default="" />

  <node name="color_detector_manager" pkg="nodelet" type="nodelet" args="manager" output="screen" />

  <node name="color_detector" pkg="nodelet" type="nodelet" args="load object_detection/ColorDetector color_detector_manager" output="screen" >
    <remap from="color_detector/image" to="$(arg camera)/image_rect_color" />
    <remap from="color_detector/camera_info" to="$(arg camera)/camera_info" />
    <rosparam file="$(find object_detection)/cfg/color_detector_params.yaml" />
    <param name="db_type" value="filesystem" />
    <param name="connection_string" value="$(find object_detection_models)/models" />
    <param name="show_images" value="$(arg show_images)" />
    <param name="queue_size" value="5" />
    <param name="models" value="$(arg models)" />
  </node>

  <node name="shape_detector_manager" pkg="nodelet" type="nodelet" args="manager" output="screen" />

  <node name="shape_detector" pkg="nodelet" type="nodelet" args="load object_detection/ShapeDetector shape_detector_manager" output="screen" >
    <remap from="shape_detector/input_detections" to="color_detector/detections" />
    <remap from="shape_detector/image" to="$(arg camera)/image_rect_color" />
    <remap from="shape_detector/camera_info" to="$(arg camera)/camera_info" />
    <rosparam file="$(find object_detection)/cfg/shape_detector_params.yaml" />
    <param name="db_type" value="filesystem" />
    <param name="connection_string" value="$(find object_detection_models)/models" />
    <param name="queue_size" value="5" />
    <param name="models" value="$(arg models)" />
  </node>

</launch>
